lib/rbvmomi/connection.rb

```diff
@@ -195,8 +195,7 @@ def obj2xml xml, name, type, is_array, o, attrs={}
       end
       xml
     rescue
-      $stderr.puts "#{$!.class} while serializing #{name} (#{type}):"
-      PP.pp o, $stderr
+      RbVmomi.logger.error("#{$!.class} while serializing #{name} (#{type}): #{o}")
```

Comment:
NOTE I wasn't able to find a good way of using PP that would return just a string, and didn't want to do `PP.pp(o, RbVmomi.logger.instance_variable_get(:@logdev)`

Reply:
`o.pretty_inspect` ?

Reply:
Nice didn't know about this one